cli/tests/pcluster/validators/test_secret_validators.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+from pcluster.aws.common import AWSClientError
+from pcluster.validators.common import FailureLevel
+from pcluster.validators.secret_validators import ArnServiceAndResourceValidator, MungeKeySecretSizeAndBase64Validator
+from tests.pcluster.validators.utils import assert_failure_level, assert_failure_messages
+
+
+@pytest.mark.parametrize(
+    "arn, region, expected_service, expected_resource, expected_response",
+    [
+        (
+            "arn:aws:valid_service:us-west-2:123456789012:valid_resource:valid_resource_name",
```

Comment:
If some ARNs are structured differently (like the ssm secrets, where component 5 is structured as `resource-type/resource-id`, are we sure the test cases implemented so far are enough?

Do we want to have a case with only `resource_type` as component 5? Do we want a case with `resource-type/resource-id` as component 5?

Reply:
This Validator for now can not handle this situation, let's leave it for future.